test/unit_tests/unit_test_math.cpp

```diff
@@ -114,6 +114,46 @@ namespace pvmath
         }
         return true;
     }
+
+    template <typename _Type>
+    bool pointProjection()
+    {
+        PointBase2D<_Type> point0( 0, 0 );
```

Comment:
What if we have random data for these 3 points as well:
```cpp
PointBase2D<_Type> point0( randomValue(), randomValue() );
PointBase2D<_Type> point1( point0.x, randomValue() );
PointBase2D<_Type> point2( randomValue(), point.y );
```

Moreover we could something like this:
```cpp
PointBase2D<_Type> pointBase( randomValue(), randomValue() );
const LineBase2D<_Type> lineX( pointBase, PointBase2D<_Type>( pointBase.x, randomValue() ) );
const LineBase2D<_Type> lineY( pointBase, PointBase2D<_Type>( randomValue(), pointBase.y ) );
```

Looks better?

Reply:
Looks interesting!

Reply:
Hmm, I've got an unexpected problem with rounding of float values. Tests with doubles are OK, but tests with floats don't pass. I'm going to solve this problem, but it might take a while.

Reply:
Could you specify details of your problem? I faced the same issue some time ago?

Reply:
Ok. I made lines to be based on random values:

```
const PointBase2D<_Type> pointBase( Unit_Test::randomFloatValue<_Type>( -1000, 1000, 0.01f ), Unit_Test::randomFloatValue<_Type>( -1000, 1000, 0.01f ) );
const PointBase2D<_Type> pointBaseX( Unit_Test::randomFloatValue<_Type>( -1000, 1000, 0.01f ), pointBase.y );
const PointBase2D<_Type> pointBaseY( pointBase.x, Unit_Test::randomFloatValue<_Type>( -1000, 1000, 0.01f ) );

const LineBase2D<_Type> lineX( pointBase, pointBaseX );
const LineBase2D<_Type> lineY( pointBase, pointBaseY );
```

After starting unit_tests I get this:
```
341 of 342 tests passed.
List of failed tests: 
math::Line2d point opposition (float)
```

I calculated the difference between real values of opposition and expected values:

```
std::cout << lineX.opposite( testPoint ).x - resultPointX.x << " " << lineX.opposite( testPoint ).y - resultPointX.y << std::endl;
std::cout << lineY.opposite( testPoint ).x - resultPointY.x << " " << lineY.opposite( testPoint ).y - resultPointY.y << std::endl;
```

So, in `math::Line2d point opposition (double)` I get:
```
0 0
0 0
0 0
0 0
...
0 0
0 0
```

But in `math::Line2d point opposition (double)` test the result a bit different:
```
[342/342] math::Line2d point opposition (float)... 
0 0
0 0


0 0
0 0


0 0
0 3.051757812e-05

FAIL
```
Or even:
```
[342/342] math::Line2d point opposition (float)... 
0 0
0 0


0 0
0 0


0 0
0 0


0.0001220703125 0
0 0

FAIL
```

Reply:
This is definitely due to limitations of float format.

Reply:
I added another comment above with suggested accuracy improvement. Could you please rerun tests after?